routes/user.go

```diff
@@ -1811,12 +1811,87 @@ type GetNotificationsRequest struct {
 	FilteredOutNotificationCategories map[string]bool
 }
 
+type GetNotificationsCountRequest struct {
+	PublicKeyBase58Check string
+}
+
+
 type GetNotificationsResponse struct {
 	Notifications       []*TransactionMetadataResponse
 	ProfilesByPublicKey map[string]*ProfileEntryResponse
 	PostsByHash         map[string]*PostEntryResponse
 }
 
+type GetNotificationsCountResponse struct {
+	NotificationsCount uint64
+}
+
+func (fes *APIServer) GetNotificationsCount(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetNotificationsCountRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetNotificationsCount: Problem parsing request body: %v", err))
+		return
+	}
+
```

Comment:
We should probably validate the JWT here to make sure that the person requesting the notifications count for the given public key is in fact the owner of that public key.  Notification count is not super sensitive PII but people might still want to keep it private.  See the referral info endpoints for an example of how to validate the JWT is signed by the correct pub key: https://github.com/deso-protocol/backend/blob/ff3b82bdeca8a469dfeb03cb09254b6daa3cbfd5/routes/referrals.go#L32-L41

Reply:
Logic that effects a user's metadata has been moved to a separate route that is JWT protected.